index.py

```diff
@@ -27,24 +27,44 @@ def restore_index(self):
             logging.info('No previous index was found')
 
     def search(self, query: str, number_of_results: int):
-        if len(query.split(' ')) > 1:
-            raise AttributeError('Only single word query\'s are supported for now')
         processed_query = self.get_tokens(query)
-        results = list()
+
+        result_stack = []
+        pending_operation = ''
         for token in processed_query:
-            pointers = self.index.get(token)
-            if not pointers:
-                continue
-            i = 0
-            for pointer in pointers:
-
-                comment = CsvHelper.read_comment(pointer)
-                results.append(comment)
-                if i > number_of_results:
-                    break
-                i += 1
-
-        return results
+            if token == 'and' or token == 'or' or token == 'not':
```

Comment:
I think it would be better to look for the uppercase 'special' tokens

Reply:
i kind of agree. but we currently do not support case sensitivity... normalization lowercases tokens. if you think it is important enough i can refactor it so that we can bypass the normalization here.

Reply:
not important for now, because we have very simple querys now